2D Crack mesh cut

lynnmunday opened the pull request:
ref #23572

lynnmunday committed:
change string compares to enums

lynnmunday committed:
remove addParams from global namespace and clean up includes

lynnmunday committed:
2D xfem mesh cutter user object
moving output of xfem cutter mesh to Outputs block object
ref #23572

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1382530/) on e74939d wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/23573/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/23573/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format b718d256f6e31b288be163b8809790dcfcb871da`

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1383010/) on 96b2112 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/23573/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1383046/) on 96b2112 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>9799cf</th>
      <th colspan=4>#23573 96b211</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>84.77%</td>
      <td>84.77%</td>
      <td>+0.00%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>83458</td>
      <td>83460</td>
      <td>+2</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>14998</td>
      <td>14996</td>
      <td>-2</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/23573/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/23573/coverage/framework)


## Modules coverage

### Tensor mechanics

<table>
  <thead>
    <tr>
      <th></th>
      <th>9799cf</th>
      <th colspan=4>#23573 96b211</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>89.21%</td>
      <td>89.21%</td>
      <td>-</td>
      <td>100.00%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>26511</td>
      <td>26511</td>
      <td>-</td>
      <td>2</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>3205</td>
      <td>3205</td>
      <td>-</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/23573/coverage_diff/tensor_mechanics)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/23573/coverage/tensor_mechanics)

### Xfem

<table>
  <thead>
    <tr>
      <th></th>
      <th>9799cf</th>
      <th colspan=4>#23573 96b211</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>81.85%</td>
      <td>82.02%</td>
      <td>+0.17%</td>
      <td>90.75%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>6348</td>
      <td>6494</td>
      <td>+146</td>
      <td>157</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>1408</td>
      <td>1424</td>
      <td>+16</td>
      <td>16</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/23573/coverage_diff/xfem)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/23573/coverage/xfem)

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/23573/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/23573/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/23573/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/23573/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/23573/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/23573/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/23573/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/23573/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/23573/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/23573/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/23573/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/23573/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/23573/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/23573/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/23573/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/23573/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/23573/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/23573/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/23573/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/23573/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/23573/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/23573/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/23573/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/23573/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/23573/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/23573/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/23573/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/23573/coverage/xfem)

</details>



This comment will be updated on new commits.

lynnmunday committed:
changed include to forward declare ref #23572

lynnmunday committed:
added base class to 2D crack mesh cut UO

lynnmunday committed:
Using fracture integral to propagate 1D crack on 2D domain using a meshcutter for xfem.  Handles multiple crack fronts.  closes #23572

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1402908/) on 1256ba4 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/23573/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/23573/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format cbdda2fe459a66ccea2d613d89b4d0f5fad0f93e`

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1405032/) on 08bfb69 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/23573/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/23573/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 1602929093c1189b311e78c04c0c7f10f0d80323`

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1406082/) on f1249d5 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/23573/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/23573/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 1f188dd3d7d67e657d0aa31546a71f0fd21d32b2`

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1409630/) on dcd9002 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/23573/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/23573/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format ae61f4e8046d6aa06bc153d0cfcf0c729c297195`

lynnmunday committed:
refactor and consolidate meshCut2D functions into base class

lynnmunday committed:
Update interaction integral vector postprocessors and CrackFrontDefinition with current xfem cutter data when xfem_update > 1
adding xfem mesh cut update test and modifying disabling hdf5 xfem exodus output
close #23572

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1418838/) on 7f144a4 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/23573/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/23573/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format ba109d0eb9dde4fa0965ce7253c956b13bf00549`

lynnmunday committed:
increasing abs_zero for the edge_crack_3d_mhs because it started diffing after setting exec_flags to nonlinear.  The increase should get rid of small values in the z-direction are nearly zero because the crack face is normal to z

lynnmunday committed:
changing preconditioner to LU to speed up simulation.  The debug version of this test was exceeding 300s due to forcing execute_on nonlinear in the vpps for interaction integrals

lynnmunday committed:
Adding documentation for MeshCut2DFractureUserObject ref #23572

lynnmunday commented:
@bwspenc It is ready to review.  Everything that is failing wasn't touched by this pr.

lynnmunday committed:
making sure kI is positive to open crack

jiangwen84 commented on modules/xfem/src/outputs/XFEMCutMeshOutput.C:
```diff
@@ -0,0 +1,68 @@
+  output << _file_base << ".e";
+
+  // Add the _000x extension to the file
+  if (_file_num > 1)
```
>= 1

jiangwen84 replied:
```>=1```

lynnmunday commented on modules/tensor_mechanics/src/actions/DomainIntegralAction.C:
```diff
@@ -32,7 +32,7 @@ InputParameters
 {
   InputParameters params = Action::validParams();
-  addCrackFrontDefinitionParams(params);
+  CrackFrontDefinition::addCrackFrontDefinitionParams(params);
```
I put addCrackFrontDefinitionParams back into the global namespace and submitted a ticket to fix it, #24335

lynnmunday committed:
Moving addCrackFrontDefinitionParams to fix failing Grizzly build

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1520588/) on d908c17 wanted to post the following:


The following coverage requirement(s) failed:
- Failed to generate `chemical_reactions` coverage rate (required: 92.0%)
- Failed to generate `combined` coverage rate (required: 37.0%)
- Failed to generate `contact` coverage rate (required: 87.0%)
- Failed to generate `electromagnetics` coverage rate (required: 95.0%)
- Failed to generate `external_petsc_solver` coverage rate (required: 85.0%)
- Failed to generate `fluid_properties` coverage rate (required: 84.0%)
- Failed to generate `fsi` coverage rate (required: 90.0%)
- Failed to generate `functional_expansion_tools` coverage rate (required: 82.0%)
- Failed to generate `geochemistry` coverage rate (required: 96.0%)
- Failed to generate `heat_conduction` coverage rate (required: 87.0%)
- Failed to generate `level_set` coverage rate (required: 88.0%)
- Failed to generate `misc` coverage rate (required: 34.0%)
- Failed to generate `navier_stokes` coverage rate (required: 77.0%)
- Failed to generate `optimization` coverage rate (required: 86.0%)
- Failed to generate `peridynamics` coverage rate (required: 78.0%)
- Failed to generate `phase_field` coverage rate (required: 85.0%)
- Failed to generate `porous_flow` coverage rate (required: 95.0%)
- Failed to generate `ray_tracing` coverage rate (required: 93.0%)
- Failed to generate `rdg` coverage rate (required: 64.0%)
- Failed to generate `reactor` coverage rate (required: 90.0%)
- Failed to generate `richards` coverage rate (required: 94.0%)
- Failed to generate `scalar_transport` coverage rate (required: 90.0%)
- Failed to generate `solid_properties` coverage rate (required: 83.0%)
- Failed to generate `stochastic_tools` coverage rate (required: 88.0%)
- Failed to generate `tensor_mechanics` coverage rate (required: 84.0%)
- Failed to generate `thermal_hydraulics` coverage rate (required: 90.0%)
- Failed to generate `xfem` coverage rate (required: 80.0%)

lynnmunday commented:
@bwspenc It is passing most of the tests now and is no longer breaking Grizzly.  Will you review it and look for how we can add or try to test nucleating new crack fronts?  The cutter uses an exodus mesh for the initial crack and that mesh is updated as it cuts.  If we don't have any cracks to begin with, we will have to create a crack mesh during nucleation.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1520588/) on d908c17 wanted to post the following:


The following coverage requirement(s) failed:
- Failed to generate `chemical_reactions` coverage rate (required: 92.0%)
- Failed to generate `combined` coverage rate (required: 37.0%)
- Failed to generate `contact` coverage rate (required: 87.0%)
- Failed to generate `electromagnetics` coverage rate (required: 95.0%)
- Failed to generate `external_petsc_solver` coverage rate (required: 85.0%)
- Failed to generate `fluid_properties` coverage rate (required: 84.0%)
- Failed to generate `fsi` coverage rate (required: 90.0%)
- Failed to generate `functional_expansion_tools` coverage rate (required: 82.0%)
- Failed to generate `geochemistry` coverage rate (required: 96.0%)
- Failed to generate `heat_conduction` coverage rate (required: 87.0%)
- Failed to generate `level_set` coverage rate (required: 88.0%)
- Failed to generate `misc` coverage rate (required: 34.0%)
- Failed to generate `navier_stokes` coverage rate (required: 77.0%)
- Failed to generate `optimization` coverage rate (required: 86.0%)
- Failed to generate `peridynamics` coverage rate (required: 78.0%)
- Failed to generate `phase_field` coverage rate (required: 85.0%)
- Failed to generate `porous_flow` coverage rate (required: 95.0%)
- Failed to generate `ray_tracing` coverage rate (required: 93.0%)
- Failed to generate `rdg` coverage rate (required: 64.0%)
- Failed to generate `reactor` coverage rate (required: 90.0%)
- Failed to generate `richards` coverage rate (required: 94.0%)
- Failed to generate `scalar_transport` coverage rate (required: 90.0%)
- Failed to generate `solid_properties` coverage rate (required: 83.0%)
- Failed to generate `stochastic_tools` coverage rate (required: 88.0%)
- Failed to generate `tensor_mechanics` coverage rate (required: 84.0%)
- Failed to generate `thermal_hydraulics` coverage rate (required: 90.0%)
- Failed to generate `xfem` coverage rate (required: 80.0%)

bwspenc commented on modules/xfem/src/userobjects/MeshCut2DUserObjectBase.C:
```diff
@@ -0,0 +1,334 @@
+    {
+
+      Point end_pt, connecting_pt;
+      // make sure normal crossed with the crack face points in the [0,0,1] direction
```
This comment isn't really describing what's going on below, is it?

lynnmunday replied:
It is but there is a better comment 10 lines down.

bwspenc commented on modules/xfem/src/userobjects/MeshCut2DUserObjectBase.C:
```diff
@@ -0,0 +1,334 @@
+      }
+
+      Point fracture_dir = end_pt - connecting_pt;
+      // crack normal is fracture direction cross tangent direction
+      // In 2D, tangent_direction is always [001] in CrackFrontDefinition.C so
+      // need to make sure [001] x normal_direction gives fracture_dir
+      // SO... fracture_dir x [001] gives the correct normal_direction
```
```suggestion
      // The crack normal is orthogonal to the crack extension direction (fracture_dir), 
      // and is defined in this implementation as the cross product of the direction of crack
      // extension with the tangent direction, which is always (0, 0, 1) in 2D.
```

bwspenc commented on modules/xfem/test/tests/solid_mechanics_basic/tests:
```diff
@@ -83,6 +83,38 @@
+    expect_err = 'timestep size must be greater than 1e-06, for crack to grow. dt=1e-07'
+    # XFEM requires --enable-unique-ids in libmesh
+    unique_id = true
+    requirement = 'The XFEM module MeshCut2DFunctionUserObject will error if the crack growth rate timestep falls below 1e-6s'
```
```suggestion
    requirement = 'The XFEM module MeshCut2DFunctionUserObject shall generate an error if the crack growth rate timestep falls below 1e-6s'
```

lynnmunday commented on modules/tensor_mechanics/include/userobjects/CrackFrontDefinition.h:
```diff
@@ -328,9 +334,9 @@ class CrackFrontDefinition : public GeneralUserObject, public BoundaryRestrictab
   /// Vector normals to a nonplanar crack described by the cutter mesh when _use_mesh_cutter = true
   std::vector<RealVectorValue> _crack_plane_normals;
-  /// Whether to treat a 3D model as 2D for computation of fracture integrals
+  /// Whether to treat the model as 2D for computation of fracture integrals
```
We talked and you are going to check this.  We might want to add a new bool for the 2d xfem cutter.

lynnmunday commented on modules/tensor_mechanics/src/actions/DomainIntegralAction.C:
```diff
@@ -415,7 +415,7 @@ DomainIntegralAction::act()
       params.set<MooseEnum>("family") = _family;
 
-      if (_treat_as_2d)
+      if (_treat_as_2d && _use_crack_front_points_provider == false)
```
You are correct about this being for 3D models.  It sets up the correct vectorpostprocessors that I needed for this new cutter so I reused it.  Maybe we can add a new bool.

lynnmunday commented on modules/xfem/include/userobjects/MeshCut2DUserObjectBase.h:
```diff
@@ -0,0 +1,76 @@
+   * assosciated current crack front node id in pair.second.  This vector is sorted on pair.first
+   * which makes the ordering of this vector the same as that used in the CrackFrontDefinition
+   */
+  std::vector<std::pair<dof_id_type, dof_id_type>> _original_and_current_front_node_ids;
```
We talked about this.  We decided the name is fine but the comment needs to be clearer

lynnmunday commented on modules/xfem/src/userobjects/MeshCut2DUserObjectBase.C:
```diff
@@ -0,0 +1,334 @@
+                                              std::vector<Xfem::CutNode> & cut_nodes) const
+{
+  // With the crack defined by a line, this method cuts a 2D elements by a line
+  // Fixme lynn Copy and paste from InterfaceMeshCut2DUserObject::cutElementByGeometry
```
I don't know.  It is a copy and paste and if we refactor one instance of it, we should refactor this as well.  Maybe they can be combined or it can be moved to a base class.  I don't think it is using any member data so maybe it can be a static.

lynnmunday committed:
Apply suggestions from code review

Co-authored-by: Ben Spencer <benjamin.spencer@inl.gov>

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1536270/) on a35f039 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/23573/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/23573/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format aac072163b0a7aaf33a7dc5807034f928df6afa4`

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1536270/) on a35f039 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/23573/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/23573/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format aac072163b0a7aaf33a7dc5807034f928df6afa4`

lynnmunday committed:
Addressing reviwer comments

lynnmunday commented on modules/xfem/test/tests/solid_mechanics_basic/edge_crack_2d_propagation_mhs.i:
It means max hoop stress growth_dir_method in the CrackMeshCut2DUserObject.  This was the same for the 3D case where I copied this from.  I'll just drop it for the 2D examples.

lynnmunday commented on modules/xfem/src/userobjects/MeshCut2DFunctionUserObject.C:
```diff
@@ -0,0 +1,83 @@
+{
+  // following logic only calls crack growth function if time changed.
+  // This deals with max_xfem_update > 1.  Error if the timestep drops below this
+  if ((_t - _time_of_previous_call_to_UO) > libMesh::TOLERANCE)
```
fixed.

lynnmunday commented on modules/xfem/test/tests/solid_mechanics_basic/tests:
```diff
@@ -83,6 +83,38 @@
+    requirement = 'The XFEM module shall represent a propagating crack in a 2D mechanics model with XFEM, where the crack is defined using a topologically surface cutting mesh by MeshCut2DFunctionUserObject and crack growth is defined by a user provided function'
+    design = 'MeshCut2DFunctionUserObject.md'
+  [../]
+  [./edge_crack_2d_propagation_error]
```
fixed

lynnmunday commented on modules/xfem/src/userobjects/MeshCut2DFunctionUserObject.C:
```diff
@@ -0,0 +1,83 @@
+  InputParameters params = MeshCut2DUserObjectBase::validParams();
+  params.addClassDescription("Creates a UserObject for a mesh cutter in 2D problems where crack "
+                             "growth is specified by functions.");
+  params.addRequiredParam<FunctionName>("function_x", "Growth function for x direction");
+  params.addRequiredParam<FunctionName>("function_y", "Growth function for y direction");
+  params.addRequiredParam<FunctionName>("function_v", "Growth speed function");
```
fixed

lynnmunday commented on modules/tensor_mechanics/src/userobjects/CrackFrontDefinition.C:
```diff
@@ -786,31 +796,51 @@ CrackFrontDefinition::updateCrackFrontGeometry()
+      rot_mat(2, _axis_2d) = 1.0;
+      if (_use_mesh_cutter)
+      {
+        rot_mat.fillRow(1, _crack_plane_normals[i]);
```
I got rid of the clear and put in an assert.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1542209/) on f1faac7 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/23573/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/23573/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format eec47de51ca5a6d40d576441e690c5db0378629e`

lynnmunday commented on modules/tensor_mechanics/test/tests/t_stress/ad_t_stress_ellip_crack_3d.i:
```diff
@@ -106,9 +106,10 @@
+
+  solve_type = NEWTON
+  petsc_options_iname = '-pc_type'
+  petsc_options_value = 'lu'
```
I fixed the execute on nonlinear and reverted this test.

lynnmunday commented on modules/tensor_mechanics/src/actions/DomainIntegralAction.C:
```diff
@@ -340,7 +340,7 @@ DomainIntegralAction::act()
 
     InputParameters params = _factory.getValidParams(uo_type_name);
-    params.set<ExecFlagEnum>("execute_on") = {EXEC_INITIAL, EXEC_TIMESTEP_END};
+    params.set<ExecFlagEnum>("execute_on") = {EXEC_INITIAL, EXEC_TIMESTEP_END, EXEC_NONLINEAR};
```
I changed it to only execute on nonlinear if it is using an xfem cutter mesh.

lynnmunday committed:
addressed more reviewer comments

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1544211/) on db59b77 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/23573/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/23573/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 395ab6da47fe9a63aab358a9ca122e2fd6dd943d`

lynnmunday commented on modules/xfem/src/userobjects/MeshCut2DUserObjectBase.C:
```diff
@@ -0,0 +1,334 @@
+MeshCut2DUserObjectBase::cutElementByGeometry(const Elem * /*elem*/,
+                                              std::vector<Xfem::CutFace> & /*cut_faces*/) const
+{
+  mooseError("Invalid method for 2D mesh cutting, must use "
```
removed the second part

lynnmunday commented on modules/xfem/src/userobjects/MeshCut2DUserObjectBase.C:
```diff
@@ -0,0 +1,334 @@
+  return false;
+}
+
+// fixme lynn
```
it was a cut and paste minor modifications.

lynnmunday commented on modules/xfem/src/userobjects/MeshCut2DUserObjectBase.C:
```diff
@@ -0,0 +1,334 @@
+                                               std::vector<Xfem::CutFace> & /*cut_faces*/) const
+{
+  mooseError("Invalid method for 2D mesh fragment cutting, must use "
+             "vector of element edges for 2D mesh fragment cutting");
```
dropped second part of this message.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1544790/) on 6004551 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/23573/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/23573/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 395ab6da47fe9a63aab358a9ca122e2fd6dd943d`

lynnmunday commented:
@bwspenc I addressed all of your comments and made the changes we talked about.

lynnmunday committed:
Apply suggestions from code review

Co-authored-by: Ben Spencer <benjamin.spencer@inl.gov>

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1548923/) on d8b994e wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/23573/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/23573/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format a7a0bdf98983abd194b86365de3c2fe72ffb3ea2`

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1548923/) on d8b994e wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/23573/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/23573/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format a7a0bdf98983abd194b86365de3c2fe72ffb3ea2`

lynnmunday committed:
Addressed reviewer comments about domainIntegralAction and knowing if vpp should be executed on nonlinear

lynnmunday commented on modules/tensor_mechanics/include/userobjects/CrackFrontPointsProvider.h:
```diff
@@ -32,4 +32,14 @@ class CrackFrontPointsProvider : public ElementUserObject
+
+protected:
+  /// bool to set if CrackFronPointsProvider derived objects use a cutter mesh
+  bool _uses_mesh;
```
I made this const.

lynnmunday commented:
@bwspenc Will you take another look at this PR.  All of the test failures are in code I didn't touch.

lynnmunday committed:
changing input file to show that code changes fix order dependence of CrackFrontDefinition user objects needing to be constructed before the crackFrontDefinition userObject is created by the domainIntegralAction

lynnmunday commented:
It was passing all of the tests but I pushed a new commit to show that the order dependence of userObject construction has been fixed.

moosebuild commented:
Job [OpenMPI](https://civet.inl.gov/job/1552654/) on 2cbcaa3 : invalidated by @bwspenc

moosebuild commented:
Job [OpenMPI](https://civet.inl.gov/job/1552654/) on 2cbcaa3 : invalidated by @bwspenc

bwspenc commented:
Failure of richards module test in OpenMPI test target is unrelated to this PR. See https://github.com/idaholab/moose/issues/24459
